### ISSUE TITLE
sbt compile keeps reformatting some Java sources

### ISSUE THREAD

**Issue by dope9967** [akka/akka#28951]
Hello,

I’m trying to compile Akka on my workspace PC using IntelliJ, trying to figure out whole setup so I can contribute productively. I had to work around some issues with IntelliJ’s build process, fell back to compiling builds with sbt shell, which is basically sbt running in a terminal. Issue is that whenever I recompile some Java sources get formatted and I can’t figure out why. Tried running sbt compile from terminal too, same result. What could be the issue?

Environment:
Windows 10
IntelliJ 2020.1
sbt 1.3.10

Commit: https://github.com/akka/akka/tree/6361bdb1533b6a180926e1eb3bfc8adf5ff79a84

List of files that get modified during compile:
modified: akka-actor-tests/src/test/java/akka/dispatch/JavaFutureTests.java
modified: akka-actor/src/main/java/akka/dispatch/AbstractBoundedNodeQueue.java
modified: akka-actor/src/main/java/akka/dispatch/AbstractMailbox.java
modified: akka-actor/src/main/java/akka/dispatch/AbstractMessageDispatcher.java
modified: akka-actor/src/main/java/akka/dispatch/AbstractNodeQueue.java
modified: akka-cluster-metrics/src/main/java/akka/cluster/metrics/protobuf/msg/ClusterMetricsMessages.java
modified: akka-cluster-sharding-typed/src/main/java/akka/cluster/sharding/typed/internal/protobuf/ShardingMessages.java
modified: akka-cluster-sharding/src/main/java/akka/cluster/sharding/protobuf/msg/ClusterShardingMessages.java
modified: akka-cluster-tools/src/main/java/akka/cluster/client/protobuf/msg/ClusterClientMessages.java
modified: akka-cluster-tools/src/main/java/akka/cluster/pubsub/protobuf/msg/DistributedPubSubMessages.java
modified: akka-cluster-typed/src/main/java/akka/cluster/typed/internal/protobuf/ClusterMessages.java
modified: akka-cluster-typed/src/main/java/akka/cluster/typed/internal/protobuf/ReliableDelivery.java
modified: akka-cluster/src/main/java/akka/cluster/protobuf/msg/ClusterMessages.java
modified: akka-distributed-data/src/main/java/akka/cluster/ddata/protobuf/msg/ReplicatedDataMessages.java
modified: akka-distributed-data/src/main/java/akka/cluster/ddata/protobuf/msg/ReplicatorMessages.java
modified: akka-docs/src/main/java/docs/ddata/protobuf/msg/TwoPhaseSetMessages.java
modified: akka-docs/src/main/java/docs/persistence/proto/FlightAppModels.java
modified: akka-multi-node-testkit/src/main/java/akka/remote/testconductor/TestConductorProtocol.java
modified: akka-persistence/src/main/java/akka/persistence/serialization/MessageFormats.java
modified: akka-protobuf/src/main/java/akka/protobuf/AbstractMessage.java
modified: akka-protobuf/src/main/java/akka/protobuf/AbstractMessageLite.java
modified: akka-protobuf/src/main/java/akka/protobuf/AbstractParser.java
modified: akka-protobuf/src/main/java/akka/protobuf/BlockingRpcChannel.java
modified: akka-protobuf/src/main/java/akka/protobuf/BlockingService.java
modified: akka-protobuf/src/main/java/akka/protobuf/BoundedByteString.java
modified: akka-protobuf/src/main/java/akka/protobuf/ByteString.java
modified: akka-protobuf/src/main/java/akka/protobuf/CodedInputStream.java
modified: akka-protobuf/src/main/java/akka/protobuf/CodedOutputStream.java
modified: akka-protobuf/src/main/java/akka/protobuf/DescriptorProtos.java
modified: akka-protobuf/src/main/java/akka/protobuf/Descriptors.java
modified: akka-protobuf/src/main/java/akka/protobuf/DynamicMessage.java
modified: akka-protobuf/src/main/java/akka/protobuf/ExtensionRegistry.java
modified: akka-protobuf/src/main/java/akka/protobuf/ExtensionRegistryLite.java
modified: akka-protobuf/src/main/java/akka/protobuf/FieldSet.java
modified: akka-protobuf/src/main/java/akka/protobuf/GeneratedMessage.java
modified: akka-protobuf/src/main/java/akka/protobuf/GeneratedMessageLite.java
modified: akka-protobuf/src/main/java/akka/protobuf/Internal.java
modified: akka-protobuf/src/main/java/akka/protobuf/InvalidProtocolBufferException.java
modified: akka-protobuf/src/main/java/akka/protobuf/LazyField.java
modified: akka-protobuf/src/main/java/akka/protobuf/LazyStringArrayList.java
modified: akka-protobuf/src/main/java/akka/protobuf/LazyStringList.java
modified: akka-protobuf/src/main/java/akka/protobuf/LiteralByteString.java
modified: akka-protobuf/src/main/java/akka/protobuf/Message.java
modified: akka-protobuf/src/main/java/akka/protobuf/MessageLite.java
modified: akka-protobuf/src/main/java/akka/protobuf/MessageLiteOrBuilder.java
modified: akka-protobuf/src/main/java/akka/protobuf/MessageOrBuilder.java
modified: akka-protobuf/src/main/java/akka/protobuf/Parser.java
modified: akka-protobuf/src/main/java/akka/protobuf/ProtocolMessageEnum.java
modified: akka-protobuf/src/main/java/akka/protobuf/RepeatedFieldBuilder.java
modified: akka-protobuf/src/main/java/akka/protobuf/RopeByteString.java
modified: akka-protobuf/src/main/java/akka/protobuf/RpcCallback.java
modified: akka-protobuf/src/main/java/akka/protobuf/RpcChannel.java
modified: akka-protobuf/src/main/java/akka/protobuf/RpcController.java
modified: akka-protobuf/src/main/java/akka/protobuf/RpcUtil.java
modified: akka-protobuf/src/main/java/akka/protobuf/Service.java
modified: akka-protobuf/src/main/java/akka/protobuf/SingleFieldBuilder.java
modified: akka-protobuf/src/main/java/akka/protobuf/SmallSortedMap.java
modified: akka-protobuf/src/main/java/akka/protobuf/TextFormat.java
modified: akka-protobuf/src/main/java/akka/protobuf/UninitializedMessageException.java
modified: akka-protobuf/src/main/java/akka/protobuf/UnknownFieldSet.java
modified: akka-protobuf/src/main/java/akka/protobuf/UnmodifiableLazyStringList.java
modified: akka-protobuf/src/main/java/akka/protobuf/Utf8.java
modified: akka-protobuf/src/main/java/akka/protobuf/WireFormat.java
modified: akka-remote-tests/src/test/java/akka/remote/artery/protobuf/TestMessages.java

**Comment by johanandren** [comment]
I don't think we have seen that before, but I also don't think anyone in the core team is using Windows so there could perhaps be some bugs in the sbt java formatting plugin going on?

Perhaps try the sbt-java-formatter plugin on a dummy project to see if it always happens or if it is strictly related to the Akka build: https://github.com/sbt/sbt-java-formatter

**Comment by dope9967** [comment]
Tried `javafmtCheckAll` , seems like those files get flagged by it. Tried to rollback to previous `sbt-java-formatter` version used by Akka, still some files get formatted. Seems like a plugin bug.

`sbt-java-formatter`  https://github.com/sbt/sbt-java-formatter/releases/tag/v0.5.1 provides option to disable formatting on compilation, maybe I could produce a PR that would allow to disable formatting on compilation with a VM option, similar to what's done in discipline plugin. Could be a simple workaround, Windows contributors could then reformat just before submitting PRs with the use of Linux subsystem.

**Comment by johanandren** [comment]
A bit messy for contributors on windows still but better than now if everyone has that problem. I don't like the idea of yet another flag but including formatting in the existing discipline toggle sounds ok to me.

**Comment by johanandren** [comment]
Figuring out what is wrong with the java formatter plugin and fixing that would be even better, but it's ofc up to you what you want to spend your time on ;)

**Comment by johanandren** [comment]
Sort of solved by #28962 by allowing to disable format on compile, but the root cause is not fixed so let's leave the issue open.
